idempotency/idempotency.go

```diff
@@ -0,0 +1,50 @@
+package idempotency
+
+import (
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+	"net/http"
+	"time"
+)
+
+const (
+	AppID      = "App-ID"
+	RequestID  = "Request-ID"
+)
+
+type BaseModel struct {
+	CreatedAt time.Time  `json:"created_at"`
+	UpdatedAt time.Time  `json:"updated_at"`
+	DeletedAt *time.Time `sql:"index" json:"deleted_at,omitempty"`
+}
```

Comment:
@ankitrev  There is a basemodel already present at PropertyMap, try to use that.